### PR TITLE
limbo: add 'features' set to testcases

### LINKED ISSUE #16: Schema: support a `tag` field that implies optionality
This will help us support testcases for optional parts of X.509 profiles, like certificate policy extensions.

## Behavior

The presence of a tag implies that the testcase's behavior is "optional", meaning:

1. If the implementation under test supports the testcase's functionality, then the test **MUST** have the expected result (i.e., must succeed or fail as specified).
2. If the implementation under test doesn't support the testcase's functionality **MUST** fail.

(2) is a little murky -- I'm not 100% sure this should be **MUST** fail, but it's probably fine to start with that (since we expect to mostly use this on things like critical extensions that aren't universally supported).

## Sketch

Something like this:

```
@testcase(tag="whatever")
```

...or maybe this:

```
@testcase
def whatever(builder: Builder): 
    builder.tag("whatever")
```

The latter is probably easier to implement, so we should start with that.

### PR DESCRIPTION
**Pull request by woodruffw**
Closes #16.

### CODE REVIEW

**Comment by woodruffw** [comment]
NB: I named this `features` instead of `tags` (as in #16) to avoid confusion around the purpose here -- `tags` felt pretty generic.

File: limbo/testcases/_core.py [hunk 1]
```diff
@@ -27,6 +28,10 @@ def __init__(self, id: str, description: str):
         self._expected_peer_name: PeerName | None = None
         self._expected_peer_names: list[PeerName] | None = None
 
+    def features(self, tags: list[str]) -> Self:
```
Comment (tnytown) [inline limbo/testcases/_core.py]: nit: rename this parameter?

Comment (woodruffw) [reply limbo/testcases/_core.py]: Yep, good call.

**Comment by woodruffw** [comment]
Thinking about it more, I think `features` should really be `list[Feature]`, where `Feature` is an enum. That'll give us the ability to document what each feature actually means.

File: limbo/testcases/_core.py [hunk 2]
```diff
@@ -27,6 +28,10 @@ def __init__(self, id: str, description: str):
         self._expected_peer_name: PeerName | None = None
         self._expected_peer_names: list[PeerName] | None = None
 
+    def features(self, feats: list[str]) -> Self:
```
Comment (tnytown) [inline limbo/testcases/_core.py]: `list[Feature]` for feats?

Comment (woodruffw) [reply limbo/testcases/_core.py]: Yep, thanks.

**Review by tnytown — APPROVED** [review]
LGTM!

**Pull request merged**